package/tests/test_integrations/test_hooks_utils.py

```diff
@@ -0,0 +1,285 @@
+
+    def dummy_func(x):
+        return x
+
+    from kedro.pipeline import node
+
+    kedro_node = node(
+        func=dummy_func,
+        inputs="input_data",
+        outputs="output_data",
+        name="test_node",
+    )
+
+    result = hash_node(kedro_node)
+
+    # Should return a consistent hash
+    assert isinstance(result, str)
+    assert len(result) > 0
+    # Should be deterministic
+    assert result == hash_node(kedro_node)
+
+
+def test_hash_node_with_string():
+    """Test hash_node function with string input - no mocking needed."""
+    test_input = "test_dataset"
+    result = hash_node(test_input)
+
+    assert isinstance(result, str)
+    assert len(result) > 0
+    # Should be deterministic
+    assert result == hash_node(test_input)
+
+
+def test_extract_file_paths():
+    """Test extract_file_paths function - no mocking needed."""
+
+    class MockDataset:
+        def __init__(self, filepath=None, _filepath=None):
+            if filepath:
+                self.filepath = filepath
+            if _filepath:
+                self._filepath = _filepath
+
+    # Test with filepath
+    dataset_with_filepath = MockDataset(filepath="/path/to/file.csv")
+    paths = extract_file_paths(dataset_with_filepath)
+    assert paths == ["/path/to/file.csv"]
+
+    # Test with _filepath
+    dataset_with_private_filepath = MockDataset(_filepath="/private/path/file.csv")
+    paths = extract_file_paths(dataset_with_private_filepath)
+    assert paths == ["/private/path/file.csv"]
+
+    # Test with both
+    dataset_with_both = MockDataset(filepath="/path1.csv", _filepath="/path2.csv")
+    paths = extract_file_paths(dataset_with_both)
+    assert paths == ["/path1.csv", "/path2.csv"]
+
+    # Test with neither
+    dataset_without_filepath = MockDataset()
+    paths = extract_file_paths(dataset_without_filepath)
+    assert paths == []
+
+
+def test_generate_timestamp_format():
+    """Test generate_timestamp function - no mocking needed."""
+    result = generate_timestamp()
+
+    # Verify the timestamp format
+    assert isinstance(result, str)
+    # Check if it matches the expected format
+    parsed_time = datetime.strptime(result, TIME_FORMAT)
+    assert parsed_time is not None
+
+    # Should be recent (within last minute)
+    now = datetime.now(tz=timezone.utc)
+    parsed_time_utc = parsed_time.replace(tzinfo=timezone.utc)
+    time_diff = abs((now - parsed_time_utc).total_seconds())
+    assert time_diff < 60  # Within 1 minute
+
+
+def test_create_dataset_event_basic():
+    """Test create_dataset_event function with basic parameters - no mocking needed."""
+    event_type = "after_dataset_loaded"
+    dataset_name = "test_dataset"
+
+    result = create_dataset_event(event_type, dataset_name)
+
+    expected = {
+        "event": event_type,
+        "dataset": dataset_name,
+        "node_id": hash_node(dataset_name),  # We know this works from previous tests
+        "status": "Available",
```

Comment:
we should use the enum here for Datasets, i think it was in another PR. but maybe not hardcode the values